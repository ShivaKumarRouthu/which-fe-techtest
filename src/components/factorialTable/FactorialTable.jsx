import React, { useState } from 'react'

import './factorialTable.css';
import Tile from '../tile/Tile';

const FactorialTable = () => {
  let numbers = [...Array(144).keys()].map(i => i + 1);

  const [selectedTile, setSelectedTile] = useState();
  const [isTileClicked, setIsTileClicked] = useState(false);

  const onTileClickedHandler = ( tileNumber ) => {
    if(selectedTile !== tileNumber) {
      setIsTileClicked(true);
      setSelectedTile(tileNumber);
    } else {
      setIsTileClicked(false);
      setSelectedTile();
    }
  }

  const isMultipleOf = (num) => {
    return (isTileClicked && (num % selectedTile === 0));
  }

  return (
    <ul className="container">
      {numbers.map((num) => <li key={num}><Tile num={num} onTileClicked={onTileClickedHandler} isSelected={isMultipleOf(num)} /></li>)}
    </ul>
  )
}

export default FactorialTable