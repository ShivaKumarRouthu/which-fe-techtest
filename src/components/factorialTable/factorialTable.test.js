import '@testing-library/jest-dom/extend-expect'

import {
  fireEvent,
  getByTestId,
  render,
  screen
} from '@testing-library/react'

import React from 'react'

import FactorialTable from './FactorialTable'


test('should render component', () => {
  const { asFragment } = render(<FactorialTable />)
  expect(asFragment()).toMatchSnapshot()
})

test('should show correct 144 items UI', () => {
 const { container } =  render(<FactorialTable />)
  const tiles = container.getElementsByClassName('tile-container');
  expect(tiles.length).toBe(144)
})

test('should set selected tile on clicking on item', () => {
  const { container } =  render(<FactorialTable />)
  const tile = screen.getByTestId('tile-item-2')
  fireEvent.click(tile)
  expect(tile).toHaveClass('selected');
})

test('should set selected tile on clicking on item in multiple', () => {
  const { container } =  render(<FactorialTable />)
  const tile = screen.getByTestId('tile-item-10')
  fireEvent.click(tile)
  const tiles = container.getElementsByClassName('selected');
  expect(tiles.length).toBe(14)
})

test('should unset selected tile on clicking on item and its multiple', () => {
  const { container } =  render(<FactorialTable />)
  const tile = screen.getByTestId('tile-item-10')
  fireEvent.click(tile)
  expect(tile).toHaveClass('selected');
  fireEvent.click(tile)
  expect(tile).not.toHaveClass('selected');
  const tiles = container.getElementsByClassName('selected');
  expect(tiles.length).toBe(0)
})