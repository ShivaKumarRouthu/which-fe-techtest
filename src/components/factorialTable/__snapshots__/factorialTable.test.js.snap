// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<DocumentFragment>
  <ul
    class="container"
  >
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-1"
        >
          1
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-2"
        >
          2
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-3"
        >
          3
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-4"
        >
          4
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-5"
        >
          5
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-6"
        >
          6
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-7"
        >
          7
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-8"
        >
          8
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-9"
        >
          9
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-10"
        >
          10
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-11"
        >
          11
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-12"
        >
          12
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-13"
        >
          13
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-14"
        >
          14
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-15"
        >
          15
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-16"
        >
          16
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-17"
        >
          17
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-18"
        >
          18
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-19"
        >
          19
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-20"
        >
          20
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-21"
        >
          21
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-22"
        >
          22
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-23"
        >
          23
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-24"
        >
          24
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-25"
        >
          25
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-26"
        >
          26
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-27"
        >
          27
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-28"
        >
          28
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-29"
        >
          29
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-30"
        >
          30
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-31"
        >
          31
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-32"
        >
          32
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-33"
        >
          33
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-34"
        >
          34
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-35"
        >
          35
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-36"
        >
          36
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-37"
        >
          37
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-38"
        >
          38
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-39"
        >
          39
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-40"
        >
          40
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-41"
        >
          41
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-42"
        >
          42
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-43"
        >
          43
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-44"
        >
          44
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-45"
        >
          45
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-46"
        >
          46
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-47"
        >
          47
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-48"
        >
          48
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-49"
        >
          49
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-50"
        >
          50
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-51"
        >
          51
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-52"
        >
          52
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-53"
        >
          53
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-54"
        >
          54
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-55"
        >
          55
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-56"
        >
          56
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-57"
        >
          57
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-58"
        >
          58
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-59"
        >
          59
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-60"
        >
          60
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-61"
        >
          61
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-62"
        >
          62
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-63"
        >
          63
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-64"
        >
          64
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-65"
        >
          65
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-66"
        >
          66
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-67"
        >
          67
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-68"
        >
          68
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-69"
        >
          69
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-70"
        >
          70
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-71"
        >
          71
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-72"
        >
          72
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-73"
        >
          73
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-74"
        >
          74
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-75"
        >
          75
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-76"
        >
          76
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-77"
        >
          77
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-78"
        >
          78
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-79"
        >
          79
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-80"
        >
          80
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-81"
        >
          81
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-82"
        >
          82
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-83"
        >
          83
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-84"
        >
          84
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-85"
        >
          85
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-86"
        >
          86
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-87"
        >
          87
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-88"
        >
          88
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-89"
        >
          89
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-90"
        >
          90
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-91"
        >
          91
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-92"
        >
          92
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-93"
        >
          93
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-94"
        >
          94
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-95"
        >
          95
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-96"
        >
          96
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-97"
        >
          97
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-98"
        >
          98
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-99"
        >
          99
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-100"
        >
          100
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-101"
        >
          101
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-102"
        >
          102
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-103"
        >
          103
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-104"
        >
          104
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-105"
        >
          105
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-106"
        >
          106
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-107"
        >
          107
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-108"
        >
          108
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-109"
        >
          109
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-110"
        >
          110
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-111"
        >
          111
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-112"
        >
          112
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-113"
        >
          113
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-114"
        >
          114
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-115"
        >
          115
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-116"
        >
          116
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-117"
        >
          117
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-118"
        >
          118
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-119"
        >
          119
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-120"
        >
          120
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-121"
        >
          121
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-122"
        >
          122
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-123"
        >
          123
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-124"
        >
          124
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-125"
        >
          125
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-126"
        >
          126
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-127"
        >
          127
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-128"
        >
          128
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-129"
        >
          129
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-130"
        >
          130
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-131"
        >
          131
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-132"
        >
          132
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-133"
        >
          133
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-134"
        >
          134
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-135"
        >
          135
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-136"
        >
          136
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-137"
        >
          137
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-138"
        >
          138
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-139"
        >
          139
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-140"
        >
          140
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-141"
        >
          141
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-142"
        >
          142
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-143"
        >
          143
        </button>
      </div>
    </li>
    <li>
      <div
        class="tile-container"
      >
        <button
          class="btn "
          data-testid="tile-item-144"
        >
          144
        </button>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
